nemo/collections/tts/models/magpietts.py

```diff
@@ -2831,6 +2835,166 @@ def setup_validation_data(self, dataset_cfg):
+        dummy_context_codes[:, :, 1] = self.context_audio_eos_id
+        batch['context_audio_codes'] = dummy_context_codes
+        batch['context_audio_codes_lens'] = torch.tensor([2], device=self.device, dtype=torch.long)
+
+    def do_tts(
+        self,
+        transcript: str,
+        language: str = "en",
+        apply_TN: bool = False,
+        temperature: float = 0.7,
+        topk: int = 80,
+        max_decoder_steps: int = 500,
+        use_cfg: bool = True,
+        cfg_scale: float = 2.5,
+    ) -> tuple:
+        """
+        Generate speech from raw text transcript.
+
+        This is a convenience method for single-utterance text-to-speech synthesis.
+        For batch processing, use `infer_batch` directly. Only supports baked context embedding
+        context injection, NO audio conditioning and text conditioning.
+        Custom voice generation is not supported by this method.
+
+        Args:
+            transcript: Raw text to synthesize.
+            language: Language code for text normalization and tokenization.
+                Supported values depend on model's tokenizer configuration.
+                Common: "en" (English), "de" (German), "es" (Spanish), etc.
+            apply_TN: Whether to apply text normalization to the transcript.
+                If True, uses nemo_text_processing for normalization.
+            temperature: Sampling temperature for token generation.
+            topk: Top-k sampling parameter.
+            max_decoder_steps: Maximum number of decoder steps.
+            use_cfg: Whether to use classifier-free guidance.
+            cfg_scale: Scale factor for classifier-free guidance.
+
+        Returns:
+            Tuple of (audio, audio_len) where:
+                audio: Generated audio waveform. Shape: (1, T_audio).
+                audio_len: Length of generated audio in samples. Shape: (1,).
+
+        Raises:
+            ValueError: If model does not have a baked context embedding.
+            ImportError: If apply_TN=True but nemo_text_processing is not installed.
+
+        Example:
+            >>> # If text does not need to be normalized
+            >>> audio, audio_len = model.do_tts("Hello, how are you today?")
+            >>>
+            >>> # If text needs to be normalized
+            >>> audio, audio_len = model.do_tts(
+            ...     "Hello, how are you today?",
+            ...     apply_TN=True,
+            ... )
+        """
+        assert (
+            self.has_baked_context_embedding
+        ), "Model does not have a baked context embedding. Please use a checkpoint with a baked context embedding."
+        # Apply text normalization if requested
+        normalized_text = transcript
+        if apply_TN:
+            try:
+                from nemo_text_processing.text_normalization.normalize import Normalizer
+
+                normalizer = Normalizer(input_case='cased', lang=language)
```

Comment:
What happens if we don't have TN for a specific language?

Reply:
Added normalizer cache and checks to skip normalization in case language does not exist in the Normalizer.